scenarios/kubernetes_bazel.py

```diff
@@ -52,15 +56,58 @@ def get_version():
 
 def main(args):
     """Trigger a bazel build/test run, and upload results."""
+    # pylint:disable=too-many-branches
+    if args.install:
+        for install in args.install:
+            if not os.path.isfile(install):
+                raise ValueError('Invalid install path: %s' % install)
+            check('pip', 'install', '-r', install)
+
     check('bazel', 'clean', '--expunge')
     res = 0
     try:
-        if args.build:
+        if args.build is not None:
             check('bazel', 'build', *args.build.split(' '))
         if args.release:
             check('bazel', 'build', *args.release.split(' '))
         if args.test:
-            check('bazel', 'test', *args.test.split(' '))
+            if args.test == 'affected':
+                base = os.getenv('PULL_BASE_SHA', '')
+                pull = os.getenv('PULL_PULL_SHA', '')
+                if not base or not pull:
+                    raise ValueError('PULL_BASE_SHA and PULL_PULL_SHA must be set!')
+
+                diff = check_output(
+                    'git', 'diff', '--name-only',
+                    '--diff-filter=d', '%s...%s' % (base, pull))
+                packages = check_output('bazel', 'query', '--noshow_progress', 'set(%s)' % diff)
+
+                if not packages:
+                    print 'No bazel packages affected.'
+                else:
+                    print 'Affected packages: %s' % packages
+                    # Build modified packages.
+                    buildables = check_output(
+                        'bazel', 'query',
+                        '--keep_going',
+                        '--noshow_progress',
+                        'kind(.*_binary, rdeps(//..., set(%s)))' % packages
+                        )
+
+                    if buildables:
+                        check('bazel', 'build', buildables)
+
+                    tests = check_output(
+                        'bazel', 'query',
+                        '--keep_going',
+                        '--noshow_progress',
+                        'kind(test, rdeps(//..., set(%s)))' % packages
+                        )
+
+                    print 'Triggering tests : %s' % tests
```

Comment:
I think this print statement and the [previous](https://github.com/kubernetes/test-infra/pull/3777/files#diff-c012d3ce1b21c59a5e03d594c01fe8c4R88) are not very helpful. This one even more so than the other, since `bazel test` will end-up showing you the list of tested target anyway.